avocado/utils/configure_network.py

```diff
@@ -92,3 +92,16 @@ def unset_ip(interface):
         return True
     except Exception as ex:
         raise NWException("ifdown fails: %s" % ex)
+
+
+def interfacewait(interface):
+    """
+    Waits for the interface link to be UP
+    """
+    for _ in range(0, 600, 5):
```

Comment:
Why these numbers? And this seems to be a busy loop, which you want to always avoid.

https://en.wikipedia.org/wiki/Busy_waiting

Reply:
Depending on what Linux driver parameter is changed (for example ethtool), Few linux driver interfaces takes 0 to 300 sec to bring up the interface.

The above code logic checks every 5 seconds if the interface is up and if the condition is met , it breaks through the loop immediately.

This method has advantage over waiting for specific generic sleep time across driver.